active-elastic-job/active-elastic-job
# Fix AWS integration testing permissions for GitHub actions (#120)

* Fix typo

* Fix spelling mistake in error message

* Document how to make mailers work

* Indent code relative to bullet points

* Drop support for rails <5 (#116)

* Drop support for rails <5

* Change ruby minimum version to 2.3.8

* Change minimum version to maintained ruby releases

* Update Travis CI config, per validation

Co-authored-by: Joey Paris <joey@leadjig.com>

* Don't try to grab credentials in development mode (#66)

* Don't try to grab credentials in development mode

* Fix

Co-authored-by: joeyparis <jparis94@gmail.com>
Co-authored-by: Joey Paris <joey@leadjig.com>

* Support local endpoint for ElasticMQ (#67)

Co-authored-by: Joey Paris <joey@leadjig.com>

* Document aws_region configuration setting

* Fix logic for detection of Docker environments in a multi-container enviroment (#111)

* Fix logic to detect AWS docker environement

* Add specs to exercise single and multi container environment scenarios

* Allow IPs in 172.17.0.x to be considered to be from Docker

Previously this was limited to just 172.17.0.1 which is overly
restrictive in a multi-container environment.

Co-authored-by: Mohamed Abdul Rasheed <mohideen@umd.edu>

* Upgrade aws-sdk to v3 (#88)

Co-authored-by: Joey Paris <joey@leadjig.com>

* Updating aws-sdk dependency (#95)

* requiring aws-sdk 3 instead of 2 to avoid dependency conflicts

* Adding note in README about SQS queue type supported

* bumping version

* fixing typo

Co-authored-by: Joey Paris <joey@leadjig.com>

* Fix integration testing for new AWS services

* #69: Set job provider ID (#70)

* #69: Set job provider ID

* Update testing to expect message_id when sending a message

Co-authored-by: Joey Paris <joey@leadjig.com>

* Adds aws_region config to client initializer (#64)

* adds aws_region config to client initializer

* Updates to support aws-sdk 3

* Remove Rails 4.2 spec integration

Co-authored-by: Joey Paris <joey@leadjig.com>

* Bump version number

* Move primary focus to Ruby 2.6

* Document nginx config change for longer running jobs

* Use gh actions (#118)

* Use github actions

* Ignore vendor folder

* Tidy up gemspec file

* Add envars for aws config

* Build against additional Rails versions

* Test against localstack

* Remove travis build

* fix formatting

* Remove duplicate source

* Remove duplicate source

* Add support for testing GitHub actions locally

Co-authored-by: John Mortlock <john.mortlock@sealink.com.au>
Co-authored-by: Joey Paris <joey@leadjig.com>

* Update references from old repo location

* Update CHANGELOG.md

* Successfully run full integration test with AWS

* Allow more flexibility in rails dependency

* Remove redundant github action test

* Attempt to fix AWS deployment

* Reference deployment environment to access github action secrets

* Disallow concurrernt deployment tests

* Temporarily disable intergration test termination

* Update Gemfile.lock

* Temporarily disable launch/updating when doing integration test

* Explicity define web/worker env values

* Don't redirect to HTTPS for the health check

* Restore full integration testing steps

* Adjust integration deployment testing to only run for pull requests from develop to master

Co-authored-by: Kevin Ceballos <kceballos@infuse.us>
Co-authored-by: Kevin Burke <kevin.burke@onemedical.com>
Co-authored-by: Jerome Dalbert <jerome.dalbert@gmail.com>
Co-authored-by: Joey Paris <joey@leadjig.com>
Co-authored-by: Daniel Orner <daniel.orner@wishabi.com>
Co-authored-by: Yoran Brondsema <yoran@sutori.com>
Co-authored-by: Stefan Magnuson <styrmis@gmail.com>
Co-authored-by: Mohamed Abdul Rasheed <mohideen@umd.edu>
Co-authored-by: Samnang Chhun <samnang.chhun@gmail.com>
Co-authored-by: Mahmoud Salem <eng.mahmoud.said@gmail.com>
Co-authored-by: Alexander Feiglstorfer <delooks@gmail.com>
Co-authored-by: John Mortlock <john.mortlock@gmail.com>
Co-authored-by: John Mortlock <john.mortlock@sealink.com.au>
